### PR TITLE
Refactor main tests with Enzyme

### PR DESCRIPTION
**Pull request by wKovacs64**
* Use Enzyme instead of ReactTestUtils to render styled components
* Test rendered components for presence of `classes` prop
* Test rendered components for presence of CSJS generated class names

### CODE REVIEW

**Comment by wKovacs64** [comment]
This would have prevented the earlier mistake of passing raw CSS text to `withStyles`. It should also make testing the tagged decorator syntax a little easier.

File: test/.setup.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * This file is used to initialize a DOM for the test environment. It MUST be
+ * loaded prior to React, Enzyme, etc., so we require it with mocha directly.
+ */
+import { jsdom } from 'jsdom';
+
+const exposedProperties = ['window', 'navigator', 'document'];
+const emptyDom = '<!DOCTYPE html><html><head></head><body></body></html>';
```
Comment (tizmagik) [inline test/.setup.js:8]: Redundant `<html>` tag here right?

**Review by tizmagik — APPROVED** [review]
Nice job, this is really great, thanks for this! 💯 

I can't merge this in because Github is acting up, I keep getting a 500 error, so feel free to merge this in yourself. 👍

File: test/.setup.js (hunk 1 above)
Comment (wKovacs64) [reply test/.setup.js:8]: Don't think so? Just a basic HTML5 structure, I think?

Comment (tizmagik) [reply test/.setup.js:8]: Ah yep sorry, don't mind me 😄

**Comment by tizmagik** [comment]
@wKovacs64 oh sorry forgot to ask before you merged this in. Does this mean we can get rid of react-addons-test-utils dep?

**Comment by wKovacs64** [comment]
I thought so initially, but Enzyme actually requires it.

**Pull request merged**